build_utils/release_tox.ini

```diff
@@ -1,74 +1,49 @@
-requires = tox-conda
+requires = tox>=4
 # The python environments to run the tests in
 envlist = pypi-py310-min,conda-py310-old-np,{pypi,conda}-{py310,py311,py312},pypisource-{py310,py312}
-# Skip the execution of setup.py as we do it with the correct version in commands_pre below
-skipsdist = True
+skip_install = True
 
 [testenv]
 changedir = tests_and_analysis/test
 test_command = python {toxinidir}/../tests_and_analysis/test/run_tests.py --report
 requirements_dir = {toxinidir}/../tests_and_analysis
+passenv =
+    EUPHONIC_VERSION
+    CC_LD
+setenv =
+    EXTRAS=[test,matplotlib,phonopy-reader,brille]
 
 # Test PyPI source distribution
 [testenv:pypisource-{py310,py312}]
-passenv = CC CC_LD LDFLAGS CPPFLAGS
-install_command = python -m pip install {opts} {packages}
-deps =
-    numpy
-    spglib  # Not interested here in whether source builds of spglib work
-    -r{[testenv]requirements_dir}/tox_requirements.txt
+passenv = CC, CC_LD, LDFLAGS, CPPFLAGS
 commands_pre =
     python -m pip install \
     --force-reinstall \
-    'euphonic[matplotlib,phonopy_reader,brille]=={env:EUPHONIC_VERSION}' \
+    'euphonic{env:EXTRAS}=={env:EUPHONIC_VERSION}' \
     --no-binary 'euphonic'
 commands = {[testenv]test_command}
 
-
 [testenv:pypi-{py310,py311,py312}]
-install_command = python -m pip install {opts} {packages}
-deps =
-    numpy
-    -r{[testenv]requirements_dir}/tox_requirements.txt
 commands_pre =
     python -m pip install \
     --force-reinstall \
-    'euphonic[matplotlib,phonopy_reader,brille]=={env:EUPHONIC_VERSION}' \
+    'euphonic{env:EXTRAS}=={env:EUPHONIC_VERSION}' \
     --only-binary 'euphonic'
 commands = {[testenv]test_command}
 
 [testenv:pypi-py310-min-win-linux]
-install_command = python -m pip install --force-reinstall {opts} {packages}
 platform =
     linux: linux
     windows: win32
 deps =
-    numpy==1.24.0
-commands_pre =
-    python -m pip install --force-reinstall \
-        -r{[testenv]requirements_dir}/minimum_euphonic_requirements.txt
-    python -m pip install --force-reinstall \
-        -r{[testenv]requirements_dir}/tox_requirements.txt
-    python -m pip install \
-    'euphonic[matplotlib,phonopy_reader,brille]=={env:EUPHONIC_VERSION}' \
-    --only-binary 'euphonic'
-commands = {[testenv]test_command}
-
-[testenv:pypi-py310-min-mac]
-install_command = python -m pip install --force-reinstall {opts} {packages}
-platform =
-    mac: darwin
-deps =
-    numpy==1.24.0
+    numpy==1.21.3
 commands_pre =
     python -m pip install --force-reinstall \
-        -r{[testenv]requirements_dir}/minimum_euphonic_requirements_mac.txt
-    python -m pip install --force-reinstall \
-        -r{[testenv]requirements_dir}/tox_requirements.txt
+        -r{toxinidir}/tests_and_analysis/minimum_euphonic_requirements.txt
     python -m pip install \
-    'euphonic[matplotlib,phonopy_reader,brille]=={env:EUPHONIC_VERSION}' \
+    'euphonic{env:EXTRAS}=={env:EUPHONIC_VERSION}' \
```

Comment:
I still think we should try the constraints file approach, may branch and play with that at some point.

Reply:
Agree, would prefer to separate from this PR as it applies to a wider range of test actions.